astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -646,6 +648,7 @@ def test_prod(self):
         with pytest.raises(u.UnitsError):
             np.prod(self.q)
 
+    @pytest.mark.filterwarnings("ignore:`product` is deprecated as of NumPy 1.25.0")
     def test_product(self):
         with pytest.raises(u.UnitsError):
             np.product(self.q)
```

Comment:
Why not replacing `np.product` with `np.prod` here like you did in other subpackages?

Reply:
I think this should still test that `np.product` works fine until it's removed by numpy, even if it's deprecated.